src/v/security/audit/audit_log_manager.cc

```diff
@@ -101,6 +108,10 @@ class audit_sink {
     /// Returns true if _client has a value
     bool is_enabled() const { return _client != nullptr; }
 
+    double percent_full() const {
+        return is_enabled() ? _client->percent_full() : -1.0;
```

Comment:
Can this be an optional or something?

Reply:
What exactly is the ask here? We could create/destroy the probe each time we toggle the audit_client, but that only happens on the client shard. Is there a way to selectively silence certain metrics (i.e. those that only apply to the client)?

Reply:
In general I have a distaste for in-band values that require special handling, especially when they end up in another system that has no mechanism to detect and handle them properly, such as Prometheus.

My question is: how would somebody interpret a negative percentage/ratio in Prometheus? It's not really meaningful here, a more appropriate mechanism would be to not report the metric where the value has no meaning, such as the semaphore/queue not existing.

Unfortunately you can only unregister a metric by destroying the `metric_group` it's attached to, which is somewhat inconvenient. It's probably easiest to have it's own metric group (which is just a wrapper around a handle). At that point, moving the metric to it's own probe that has a lifetime that correlates with the client probably makes most sense.

Reply:
Ah yeah, separate probe. That makes sense, thanks.